http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -1104,19 +1238,19 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
     }
   }
 
-  private[util] def maybeCreateWsmIdentityAndDatabase(app: App,
-                                                      workspaceId: WorkspaceId,
-                                                      landingZoneResources: LandingZoneResources,
-                                                      namespace: KubernetesNamespace
+  private[util] def maybeCreateWsmIdentityAndDatabases(app: App,
+                                                       workspaceId: WorkspaceId,
+                                                       landingZoneResources: LandingZoneResources,
+                                                       namespace: KubernetesNamespace
   )(implicit
     ev: Ask[F, AppContext]
-  ): F[(Option[ServiceAccountName], Option[String])] = {
-    val databaseConfigEnabled = app.appType match {
-      case AppType.Wds => config.wdsAppConfig.databaseEnabled
-      case _           => false
+  ): F[(Option[ServiceAccountName], Option[Map[String, String]])] = {
+    val shouldCreateLZDatabases = app.appType match {
+      case AppType.Wds          => landingZoneResources.postgresServer.isDefined && config.wdsAppConfig.databaseEnabled
+      case AppType.WorkflowsApp => true
```

Comment:
New app types should always be deployed with workload identity and LZ postgres, as they don't have the ability to create postgres deployments via helm

Reply:
Agree about the app requiring LZ-postgres (we are not trying to support old LZs with no postgres server).

Idly wondering what would happen if you deployed `WORKFLOWS_APP` in an old LZ _without_ postgres -- I assume it would blow up somehow.

Reply:
I'm not sure how to easily test this but I think it would error out in WSM during the attempted database creation.

Reply:
Is there an easy way to find out? Might be nice (for us in the near-future!) to fail this early rather than late in an inappropriate LZ

Reply:
If anyone has access to an old LZ in bee/dev they could grant me access too I would be more than happy to test it out (though on dev would have to be a post-merge test)